src/rspq/rspq.c

```diff
@@ -506,15 +515,22 @@ static void rspq_crash_handler(rsp_snapshot_t *state)
 static void rspq_assert_invalid_command(rsp_snapshot_t *state)
 {
     rsp_queue_t *rspq = (rsp_queue_t*)state->dmem;
+
+    // Get overlay index a name. Be defensive against DMEM corruptions.
+    int ovl_idx = rspq->current_ovl / sizeof(rspq_overlay_t);
+    const char *ovl_name = "?";
+    if (ovl_idx < RSPQ_MAX_OVERLAY_COUNT && rspq_overlay_ucodes[ovl_idx])
+        ovl_name = rspq_overlay_ucodes[ovl_idx]->name;
+
     uint32_t dmem_buffer = RSPQ_DEBUG ? 0x140 : 0x100;
     uint32_t cur = dmem_buffer + state->gpr[28];
-    printf("Invalid command\nCommand %02x not found in overlay %02x\n", state->dmem[cur], rspq->current_ovl / sizeof(rspq_overlay_t));
+    printf("Invalid command\nCommand %02x not found in overlay %s (0x%01x)\n", state->dmem[cur], ovl_name, ovl_idx);
```

Comment:
Could add an additional check for internal commands here, so it says "builtin" or something instead of "?".

Reply:
Actually since the command is not valid anyway maybe not.

Reply:
It still makes sense once it's moved into a separate function. Done.